site/src/pages/docs/advanced-streaming-backpressure.mdx

```diff
@@ -0,0 +1,129 @@
+    response.write(produceChunk(2));
+    ... // Write more chunks until we send all chunks.
+    // Call close() to end the response.
+    response.close();
+    return response;
+}
+
+...
+private HttpData produceChunk(int index) {
+    // Divide the file by the pre-defined chunk size(e.g. 8192 bytes)
+    // and read it using index.
+    // If index is 0, 0 to 8192 bytes from the file is read.
+    // If index is 1, 8193 to 16384 is read and so on.
+}
+```
+Now we can send the streaming response to the client. However, this has a problem. Even though it divides
+the file, it loads all chunks into the memory before each chunk is sent to the client,
+so the server will get the `OutOfMemoryError`. To solve it, we have to implement backpressure using
+<type://StreamWriter#whenConsumed()>:
+```java
+sb.service("/big_file.dat", (ctx, req) -> {
+    HttpResponseWriter response = HttpResponse.streaming();
+    response.write(ResponseHeaders.of(200));
+    response.whenConsumed().thenRun(() -> {
+        // Produce the first chunk when the ResponseHeaders is
+        // written to the socket.
+        response.write(produceChunk(0));
+        response.whenConsumed().thenRun(() -> {
+            // Produce the second chunk when the first chunk is
+            // written to the socket.
+            response.write(produceChunk(1));
+            ...
+        });
+    });
+    return response;
+});
+```
+
+<type://StreamWriter#whenConsumed()> returns a `CompletableFuture` that is complete when the chunk,
+which is written to the <type://HttpResponseWriter>, is finally written to the socket. So, you can add
+the next task by adding a callback (`thenRun()` in the example). We produced the next chunk using callback
+in the example.
+
+<Tip>
+
+See [Let’s Play with Reactive Streams on Armeria - 2](https://engineering.linecorp.com/en/blog/reactive-streams-armeria-2/)
+to understand how the network layer is involved in backpressure.
+
+</Tip>
+
+Of course, we need to use recursion not to implement the callback endlessly.
+```java
+sb.service("/big_file.dat", (ctx, req) -> {
+    HttpResponseWriter response = HttpResponse.streaming();
+    response.write(ResponseHeaders.of(200));
+    streamingResponse(response, 0);
+    return response;
+});
+
+private void streamingResponse(HttpResponseWriter response, int index) {
+    if (isEndOfFile() {
+        // Close the response when we send all chunks.
+        response.close();
+        return;
+    }
+    response.whenConsumed().thenRun(() -> {
+        if (response.tryWrite(produceChunk(index))) {
+            streamingResponse(response, index + 1);
+        } else {
+            // The response is completed unexpectedly.
+        }
+    });
+}
+```
+So far we have implemented the simplified version of <type://HttpFile>. Now, we can implement streaming
+responses using backpressure from any other sources (e.g. database) by simply changing the `produceChunk()`
+method to fetch data from the sources.
```

Comment:
Additionally, how about addressing a way to use the existing `Publihser` with `HttpResponse.of(headers, Publisher)` just in case users want to interop with other Reactive Streams libraries.